controllers/kserve_ray_tls_controller.go

```diff
@@ -0,0 +1,404 @@
+		hasChanged = true
+		log.V(1).Info("Delta found", "remove", existingConfigMapResource.GetName(), "namespace", existingConfigMapResource.Namespace)
+		if err = r.client.Delete(ctx, existingConfigMapResource); err != nil {
+			return err
+		}
+	}
+
+	if !hasChanged && !noMultiNodeSrExistInNs {
+		log.V(1).Info("No delta found", "name", desiredConfigMapResource.GetName(), "namespace", desiredConfigMapResource.Namespace)
+	}
+
+	return nil
+}
+
+func shouldAddRayConfigMap(existingConfigMap *corev1.ConfigMap, noMultiNodeSrExistInNs bool) bool {
+	return !noMultiNodeSrExistInNs && utils.IsNil(existingConfigMap)
+}
+func isUpdatedRayConfigMap(desiredConfigMap *corev1.ConfigMap, existingConfigMap *corev1.ConfigMap) bool {
+	return utils.IsNotNil(existingConfigMap) && !reflect.DeepEqual(desiredConfigMap.Data, existingConfigMap.Data)
+}
+func shouldDeleteRayConfigMap(existingConfigMap *corev1.ConfigMap, noMultiNodeSrExistInNs bool) bool {
+	return utils.IsNotNil(existingConfigMap) && noMultiNodeSrExistInNs
+}
+
+// reconcileRayCACertSecret watch ray-ca-cert secret in the cluster
+// and it will create/update/delete ray-ca-cert secret in the namespace where multinode ServingRuntime created
+func (r *KServeRayTlsReconciler) reconcileRayCACertSecret(ctx context.Context, log logr.Logger, ctrlNs string, targetNs string, noMultiNodeSrExistInNs bool) error {
+	// When original secret is updated, it does not need to reconcile
+	if ctrlNs == targetNs {
+		return nil
+	}
+	log.Info("Reconciling Ray CA Cert Secret", "name", constants.RayCATlsSecretName, "namespace", targetNs)
+	srcSecret := &corev1.Secret{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: constants.RayCATlsSecretName, Namespace: ctrlNs}, srcSecret)
+	if err != nil {
+		return err
+	}
+
+	// Create Desired resource
+	desiredSecretResource, err := r.createDesiredSecretResource(targetNs, srcSecret)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingSecretResource := &corev1.Secret{}
+	err = r.client.Get(ctx, types.NamespacedName{Name: constants.RayCATlsSecretName, Namespace: targetNs}, existingSecretResource)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			existingSecretResource = nil
+		} else {
+			return err
+		}
+	}
+
+	// Process Delta
+	if err = r.processDeltaSecret(ctx, log, desiredSecretResource, existingSecretResource, noMultiNodeSrExistInNs); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *KServeRayTlsReconciler) createDesiredSecretResource(destNs string, sourceSecret *corev1.Secret) (*corev1.Secret, error) {
+	desiredSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourceSecret.Name,
+			Namespace: destNs,
+			Labels: map[string]string{
+				"opendatahub.io/managed":       "true",
+				"app.kubernetes.io/name":       "odh-model-controller",
+				"app.kubernetes.io/component":  "kserve",
+				"app.kubernetes.io/part-of":    "odh-model-serving",
+				"app.kubernetes.io/managed-by": "odh-model-controller",
```

Comment:
isn't owner reference needed on the configMaps?

Reply:
it gives more information so I does not hurt .

Reply:
what do you mean?

Reply:
:) It is recommended labels by rhoai team. https://github.com/opendatahub-io/odh-model-controller/pull/221/files#r1655337054
based on this, I just added it but it might not need this for manifests though. 
But it is just more information about the object so it should not be an issue.